hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -145,7 +182,74 @@ private static AutoField getSourceField(Query query) {
                 throw new UncheckedIOException(e);
             }
         } else {
-            return new NoopField();
+            return null;
```

Comment:
trivial wrapper objects (eg. value string) aren't particularly high on my hit list at that scale/ longevity, and there's a strong chance the jvm eliminates it as dead code or suppresses the allocation via escape analysis.